zerver/tests/test_message_fetch.py

```diff
@@ -2671,6 +2675,92 @@ def _update_tsvector_index(self) -> None:
             """
             )
 
+    def test_get_visible_messages_with_narrow_with(self) -> None:
+        hamlet = self.example_user("hamlet")
+        cordelia = self.example_user("cordelia")
+        othello = self.example_user("othello")
+        self.login("hamlet")
+
+        # Test narrow with accessible stream message operand.
+        self.subscribe(self.example_user("hamlet"), "Scotland")
+
+        message_ids = [
+            self.send_stream_message(self.example_user("iago"), "Scotland") for i in range(5)
+        ]
+
+        narrow = [
+            dict(operator="channel", operand="Scotland"),
+            dict(operator="topic", operand="test"),
+            dict(operator="with", operand=f"{message_ids[0]}"),
+        ]
+        self.message_visibility_test(narrow, message_ids, 2)
+
+        # Test narrow with inaccessible stream message operand.
+        narrow = [
+            dict(operator="channel", operand="Scotland"),
+            dict(operator="topic", operand="test"),
+            dict(operator="with", operand="999999"),
+        ]
+        self.message_visibility_test(narrow, message_ids, 2)
+
+        # Test narrow with private message
```

Comment:
Be very explicit here: Something like:

# Test `with` operator is ineffective when targeting a direct message we did not receive.

Reply:
updated!